src/libraries/System.Runtime.CompilerServices.Unsafe/ref/System.Runtime.CompilerServices.Unsafe.cs

```diff
@@ -9,9 +9,11 @@ namespace System.Runtime.CompilerServices
     public static partial class Unsafe
     {
         public static ref T AddByteOffset<T>(ref T source, System.IntPtr byteOffset) { throw null; }
+        public static ref T AddByteOffset<T>(ref T source, System.UIntPtr byteOffset) { throw null; }
```

Comment:
```suggestion
        public static ref T AddByteOffset<T>(ref T source, nuint byteOffset) { throw null; }
```

The approved API has `nuint` in the signature (multiple places).

You will need to create an internal attribute in the .il file to make it work.

Reply:
You can run ildasm on C# project that uses nuint to see what the attribute looks like.

Reply:
@huoyaoyuan - you can see the reasoning for why we specifically chose `nuint` in the issue:

https://github.com/dotnet/runtime/issues/36232#issuecomment-626875185

> I would recommend changing the parameter types in the proposal to be explicitly nuint instead of UIntPtr. This will also help ensure the compiler puts the right [NativeInteger] annotation on the parameters.

Reply:
I copied `nuint` attribute definitions from [sharplab](https://sharplab.io/#v2:EYLgZgpghgLgrgJwgZwLQDlYEsBuECSAdjADQwhYA2APgAIBMAjALABQtAzAAQNcDCXAN5suont1qMAbDwAsXALIAKLMS4APEl0JxVMLlgCUQkWIC+bM0A==) . How can it be verified? The sharplab branch is allowing implicit conversion between `nuint` and `UIntPtr`.

Reply:
I do not think you need to verify it in any special way. Seeing `nuint` in the ref file and green build should be good enough.

The implicit conversions do not work the same way for UIntPtr and nuint. For example, this won't compile with UIntPtr:
```
using System;
public class C {
    public static void M(UIntPtr i) { // compiles fine when UIntPtr is changed to nuint
    }
    public static void N() {
        M(0);
    }
}
```